### PR TITLE
Core/ts version task 209 market order e2e test

### PR DESCRIPTION
**Pull request by franken5tein**
## What does this MR do?

Added market order e2e test

## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [x] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: src/test/e2e/market_order-e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,903 @@
...
+    expect((secondOrderError.response?.data as any).data[0].message).toContain(
+      "Duplicated client order id",
+    );
+
+    mockedGetOrderExpirationThreshold.mockRestore();
+    mockedCustomCheck.mockRestore();
+    mockedGetUserPossession.mockRestore();
+  });
+
+  it("Should rejected the order if the orderbook is empty", async () => {
```
Comment (hypersolo) [inline src/test/e2e/market_order-e2e.test.ts:823]: should cancel ...

File: src/test/e2e/market_order-e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,903 @@
...
+    expect(res.data.success).toBe(true);
+    expect(createdOrder.network).toEqual("ethereum_goerli");
+    expect(createdOrder.id).toEqual(1);
+    expect(createdOrder.market).toEqual(createOrderPayload.market);
+    expect(createdOrder.side).toEqual(createOrderPayload.side);
+    expect(+createdOrder.price).toEqual(+createOrderPayload.price);
+    expect(new Decimal(createdOrder.base_quantity)).toEqual(
+      math.div(createOrderPayload.amount, math.pow(10, 18)),
+    );
+    // expect(createdOrder.quote_quantity).toEqual(createOrderPayload.market);
```
Comment (hypersolo) [inline src/test/e2e/market_order-e2e.test.ts:736]: pointless comment bro :love_letter:

**Comment by hypersolo** [comment]
do we cover the scenario in which we create some fills?

File: src/test/e2e/market_order-e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,1135 @@
...
+    const createOrderPayload = {
+      tx: await signOrder(order, wallet.privateKey),
+      market: "invalid-invalid",
+      amount: 1000100000000000000,
+      price: "1600",
+      side: "buy",
+      type: "market",
+    };
+
+    const invalidNetworkError = await getError<AxiosError>(async () => {
```
Comment (GreatGaryCooper) [inline src/test/e2e/market_order-e2e.test.ts:383]: This should be `invalidMarketError`

File: src/test/e2e/market_order-e2e.test.ts [hunk 4]
```diff
@@ -0,0 +1,1135 @@
...
+      amount: 1000100000000000000,
+      price: "1600",
+      side: "buy",
+      type: "market",
+    };
+
+    createOrderPayload.tx.signature =
+      createOrderPayload.tx.signature?.concat("dskfndshjf");
+
+    const invalidNetworkError = await getError<AxiosError>(async () => {
```
Comment (GreatGaryCooper) [inline src/test/e2e/market_order-e2e.test.ts:430]: `invalidSignatureError`

File: src/test/e2e/market_order-e2e.test.ts [hunk 5]
```diff
@@ -0,0 +1,1135 @@
...
+    const createOrderPayload = {
+      tx: await signOrder(order, wallet.privateKey),
+      market: "WETH-DAI",
+      amount: 1000100000000000000,
+      price: "1600",
+      side: "buy",
+      type: "market",
+    };
+
+    const invalidNetworkError = await getError<AxiosError>(async () => {
```
Comment (GreatGaryCooper) [inline src/test/e2e/market_order-e2e.test.ts:486]: `invalidMaxFeeRatioError`

File: src/test/e2e/market_order-e2e.test.ts [hunk 6]
```diff
@@ -0,0 +1,1135 @@
...
+    const createOrderPayload = {
+      tx: await signOrder(order, wallet.privateKey),
+      market: "WETH-DAI",
+      amount: 1000100000000000000,
+      price: "1610",
+      side: "buy",
+      type: "market",
+    };
+
+    const invalidNetworkError = await getError<AxiosError>(async () => {
```
Comment (GreatGaryCooper) [inline src/test/e2e/market_order-e2e.test.ts:528]: `invalidPriceError`

File: src/test/e2e/market_order-e2e.test.ts [hunk 7]
```diff
@@ -0,0 +1,1135 @@
...
+          headers: { Authorization: `Bearer ${jwtToken}` },
+        },
+      );
+    });
+
+    expect(invalidNetworkError.response?.status).toEqual(400);
+    expect((invalidNetworkError.response?.data as any).success).toBe(false);
+    expect(
+      (invalidNetworkError.response?.data as any).data[0].message,
+    ).toContain("Invalid request parameters");
```
Comment (GreatGaryCooper) [inline src/test/e2e/market_order-e2e.test.ts:668]: Also here if possible.

**Review by GreatGaryCooper — CHANGES_REQUESTED** [review]

File: src/test/e2e/market_order-e2e.test.ts [hunk 8]
```diff
@@ -0,0 +1,1135 @@
...
+        return Promise.resolve(0);
+      });
+
+    const mockedCustomCheck = jest
+      .spyOn(provider!, "customCheck")
+      .mockImplementation(() => {
+        return Promise.resolve(true);
+      });
+
+    const insufficientBalanceError = await getError<AxiosError>(async () => {
```
[the remark this answers is not in the document]
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts]: Done!

File: src/test/e2e/market_order-e2e.test.ts (hunk 4 above)
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts]: Done!

File: src/test/e2e/market_order-e2e.test.ts (hunk 3 above)
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts]: Done!

File: src/test/e2e/market_order-e2e.test.ts (hunk 5 above)
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts]: Done!

File: src/test/e2e/market_order-e2e.test.ts (hunk 6 above)
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts]: Done!

File: src/test/e2e/market_order-e2e.test.ts [hunk 9]
```diff
@@ -0,0 +1,1135 @@
...
+    const createOrderPayload = {
+      tx: await signOrder(order, wallet.privateKey),
+      market: "WETH-DAI",
+      amount: 10000000000000,
+      price: "1600",
+      side: "buy",
+      type: "market",
+    };
+
+    const invalidNetworkError = await getError<AxiosError>(async () => {
```
[the remark this answers is not in the document]
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts]: Done!

File: src/test/e2e/market_order-e2e.test.ts [hunk 10]
```diff
@@ -0,0 +1,1135 @@
...
+          headers: { Authorization: `Bearer ${jwtToken}` },
+        },
+      );
+    });
+
+    expect(invalidNetworkError.response?.status).toEqual(400);
+    expect((invalidNetworkError.response?.data as any).success).toBe(false);
+    expect(
+      (invalidNetworkError.response?.data as any).data[0].message,
+    ).toContain("Invalid request parameters");
```
[the remark this answers is not in the document]
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts:626]: Done!

File: src/test/e2e/market_order-e2e.test.ts (hunk 7 above)
Comment (franken5tein) [reply src/test/e2e/market_order-e2e.test.ts:671]: Done!

**Review by hypersolo — APPROVED** [review]

**Review by GreatGaryCooper — APPROVED** [review]
Awesome bro, thanks 🌹

**Pull request merged**